### DISCUSSION TITLE
How do I add an IFrame Block?

### DISCUSSION THREAD

**Discussion by Fyndery** [GrapesJS/grapesjs#4502] [Q&A] [answered]
Hi! I want to let my users add Iframe-Content to their page that they create via GrapeJS. For this I want to add an IFrame-Block to the editor, but I'm struggling to get it to work. Can anybody help?

I tried this, but it only works for older versions of GrapeJS:

```
editor.DomComponents.addType("iframe", {
  isComponent: el => el.tagName === "IFRAME",
  model: {
    defaults: {
      type: "iframe",
      traits: [
        {
          type: "text",
          label: "src",
          name: "src"
        }
      ]
    }
  }
});

editor.BlockManager.add("iframe", {
  label: "iframe",
  type: "iframe",
  content: "<iframe> </iframe>",
  selectable: true
});
```

**Comment by artf** [comment, marked as answer]
Hey @influenceme there is, unfortunately, a bug for iframes right now, it was already fixed https://github.com/artf/grapesjs/commit/cbede898aaa6e1ba592e42d66147aed88179e1e1 and ready for the next release

**Reply by Fyndery** [reply]
Thank you for the quick response! I was going crazy, trying to figure it out.
